google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/SnapshotDefinition.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import com.google.api.client.util.DateTime;
+import com.google.api.core.BetaApi;
+import com.google.api.services.bigquery.model.Table;
+import com.google.auto.value.AutoValue;
+import com.google.common.annotations.VisibleForTesting;
+import javax.annotation.Nullable;
+
+@AutoValue
+@BetaApi
+public abstract class SnapshotDefinition extends TableDefinition {
+
+  private static final long serialVersionUID = 2113445776046717526L;
+
+  @AutoValue.Builder
+  public abstract static class Builder
+      extends TableDefinition.Builder<SnapshotDefinition, Builder> {
+
+    public abstract Builder setBaseTableId(TableId baseTableId);
+
+    public abstract Builder setSnapshotTime(DateTime dateTime);
```

Comment:
This seems odd to me.  Does AutoValue deal with the conversion from DateTime back and forth to an RFC3339 string?  I don't see logic in the fromPb/toPb either.

Reply:
Updated this class to use String for datetime